### ISSUE TITLE
Uses real name, not linked name, when running on softlinked files

### ISSUE THREAD

**Issue by karinlag** [sanger-pathogens/Roary#391]
It appears that roary will use the real file names when running on softlinked files (that is, linked with ln -s). I frequently make pipelines where I have unique sample directories, but where file names are the same, and then softlink files between directories for various analysis programs. Due to roary's behavior I am thus not able to do this. Not sure if this behavior is intended, or just an so far undiscovered edge case.

<pre>
2018/03/29 18:11:03 Fixing input GFF files
2018/03/29 18:11:06 Input file contains duplicate gene IDs, attempting to fix by adding a unique suffix.  New GFF in the fixed_input_files directory.  /work/projects/nn9305k/home/karinlag/2018_bioinfvi/PRJEB13885/prokka/DTU2014_min500/ecoli_03112018.gff 
2018/03/29 18:11:08 Input file contains duplicate gene IDs, attempting to fix by adding a unique suffix.  New GFF in the fixed_input_files directory.  /work/projects/nn9305k/home/karinlag/2018_bioinfvi/PRJEB13885/prokka/F159_min500/ecoli_03112018.gff 
2018/03/29 18:11:12 Input file contains duplicate gene IDs, attempting to fix by adding a unique suffix.  New GFF in the fixed_input_files directory.  /work/projects/nn9305k/home/karinlag/2018_bioinfvi/PRJEB13885/prokka/F168_min500/ecoli_03112018.gff 
2018/03/29 18:11:15 Input file contains duplicate gene IDs, attempting to fix by adding a unique suffix.  New GFF in the fixed_input_files directory.  /work/projects/nn9305k/home/karinlag/2018_bioinfvi/PRJEB13885/prokka/F27_min500/ecoli_03112018.gff 
2018/03/29 18:11:18 Input file contains duplicate gene IDs, attempting to fix by adding a unique suffix.  New GFF in the fixed_input_files directory.  /work/projects/nn9305k/home/karinlag/2018_bioinfvi/PRJEB13885/prokka/S19_min500/ecoli_03112018.gff 
2018/03/29 18:11:18 Extracting proteins from GFF files
Extracting proteins from /work/projects/nn9305k/home/karinlag/2018_bioinfvi/PRJEB13885/prokka/14042624_min500/ecoli_03112018.gff
Extracting proteins from fixed_input_files/ecoli_03112018.gff
Extracting proteins from fixed_input_files/ecoli_03112018.gff
Extracting proteins from fixed_input_files/ecoli_03112018.gff
Extracting proteins from fixed_input_files/ecoli_03112018.gff
Extracting proteins from fixed_input_files/ecoli_03112018.gff
Combine proteins into a single file
</pre>

**Comment by tseemann** [comment]
As an aside, Prokka generates unique `locus_tag` since 1.12 based on md5sum of the contigs, to avoid this.  Are you providing `--locustag` or using an old Prokka ?

**Comment by tseemann** [comment]
SO you did
```
ln -s /work/projects/nn9305k/home/karinlag/2018_bioinfvi/PRJEB13885/prokka/F159_min500/ecoli_03112018.gff 
DTU2014.gff
```
and passed `DTU2014.gff` to Roary, but it used the remote name?

**Comment by tseemann** [comment]
I notice Roary uses `fileparse()` from http://perldoc.perl.org/File/Basename.html which alludes to some "quirks". 

I wonder if this function resolves the symlinks out?

The recommend using `File::Spec->splitdir()`

**Comment by karinlag** [comment]
@tseemann Yep, your link command in the one above is what I used to link in the files. 
Note, when I ran roary, I used *.gff in the command line, in case that makes any difference.

**Comment by tseemann** [comment]
I also think this might be the cause of my recent failed tests on a new pipeline which does the same symlinking thing.
